### PR TITLE
create employee method

### PR DESCRIPTION
**Pull request by bwruan**

### CODE REVIEW

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs [hunk 1]
```diff
@@ -17,5 +17,24 @@ public async Task<List<Employee>> GetAllEmployees()
                 return await context.Employees.Select(e => new Employee() { FirstName  = e.FirstName, LastName = e.LastName, Role = e.Role, Username = e.Username, Active = e.Active }).ToListAsync();
             }
         }
+
+        public async Task CreateEmployee( string firstName, string lastName, string username, byte[] password, string role)
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs]: Let's add an extra parameter Active. But set the default of Active to true in the parameters.

That way, methods calling this create method will not need to pass in Active as parameter if they want to create an employee with true.

At the same time, by creating the Active parameter, we give ourselves the options in the future to call this method and pass in false as input if we want to create an employee whose initial state is not active

**Review by jruan1580 — CHANGES_REQUESTED** [review]
Address issue mentioned here.

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs (hunk 1 above)
Comment (bwruan) [reply EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs]: Done

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs [hunk 2]
```diff
@@ -17,5 +15,25 @@ public async Task<List<Employee>> GetAllEmployees()
...
+            using (var context = new RofSchedulerContext())
+            {
+                var employee = new Employee()
+                {
+                    FirstName = firstName,
+                    LastName = lastName,
+                    Username = username,
+                    Password = password,
+                    Role = role,
+                    Active = true
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs]: no this is not how u set default.

what if in future, i want to pass in false. this is hard coded to true.

**Review by jruan1580 — CHANGES_REQUESTED** [review]
fix, not right

**Comment by bwruan** [comment]
Fixed

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs [hunk 3]
```diff
@@ -17,5 +15,24 @@ public async Task<List<Employee>> GetAllEmployees()
                 return await context.Employees.Select(e => new Employee() { FirstName  = e.FirstName, LastName = e.LastName, Role = e.Role, Username = e.Username, Active = e.Active }).ToListAsync();
             }
         }
+
+        public async Task CreateEmployee( string firstName, string lastName, string username, byte[] password, string role, bool active = true)
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs:19]: right direction. assigning it default in parameter is good.

but now, look at how you are instantiating the employee. does it even take that field?

**Review by jruan1580 — CHANGES_REQUESTED** [review]

**Comment by bwruan** [comment]
Done

**Review by jruan1580 — APPROVED** [review]

**Pull request merged**